### PR TITLE
Implemented MEArecIO

### PR DESCRIPTION
**Pull request by alejoe91**

### CODE REVIEW

**Comment by samuelgarcia** [comment]
You need to add mearec in circleci/requirements_testing.txt
The problem we have is that merarec will installed (due to the circular dependancy) and maybe it will install the released version of neo!!

maybe we have to reverse this 2 lines in circleci/config.yml
```yaml
            pip install -r .circleci/requirements_testing.txt
            pip install .
```

File: neo/rawio/mearecrawio.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            wf_sampling_rate = self._sampling_rate
+            unit_channels.append((unit_name, unit_id, wf_units, wf_gain,
+                                  wf_offset, wf_left_sweep, wf_sampling_rate))
+        unit_channels = np.array(unit_channels, dtype=_unit_channel_dtype)
+
+        event_channels = []
+        event_channels = np.array(event_channels, dtype=_event_channel_dtype)
+
+        # fille into header dict
+        # This is mandatory!!!!!
```
Comment (samuelgarcia) [inline neo/rawio/mearecrawio.py:96]: Remove this line.

**Comment by samuelgarcia** [comment]
Can you add yourself+affiliation in `doc/source/authors.rst` ?
And fix the 2 pep8 stuff ?

File: neo/rawio/mearecrawio.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        return spike_timestamps.astype(dtype)
+
+    def _get_spike_raw_waveforms(self, block_index, seg_index, unit_index, t_start, t_stop):
+        return None
+
+    def _event_count(self, block_index, seg_index, event_channel_index):
+        return None
+
+    def _get_event_timestamps(self, block_index, seg_index, event_channel_index, t_start, t_stop):
+        return np.array([]), np.array([]), np.array([])
```
Comment (samuelgarcia) [inline neo/rawio/mearecrawio.py:159]:
```suggestion
    def _get_event_timestamps(self, block_index, seg_index, event_channel_index, t_start, t_stop):
        return None
```

File: neo/rawio/mearecrawio.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+    def _rescale_event_timestamp(self, event_timestamps, dtype):
+        event_times = event_timestamps.astype(dtype)
+        return event_times
```
Comment (samuelgarcia) [inline neo/rawio/mearecrawio.py:163]:
```suggestion
    def _rescale_event_timestamp(self, event_timestamps, dtype):
        return None
```

File: neo/rawio/mearecrawio.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+    def _rescale_epoch_duration(self, raw_duration, dtype):
+        # really easy here because in our case it is already seconds
+        durations = raw_duration.astype(dtype)
+        return durations
```
Comment (samuelgarcia) [inline neo/rawio/mearecrawio.py:168]:
```suggestion
        return None
```

File: neo/test/rawiotest/test_mearecrawio.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+"""
+Tests of neo.rawio.examplerawio
```
Comment (samuelgarcia) [inline neo/test/rawiotest/test_mearecrawio.py:2]: No

File: neo/test/rawiotest/test_mearecrawio.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+If you have problems, do not hesitate to ask help github (prefered)
+of neuralensemble list.
+
+Note that same mechanism is used a neo.io API so files are tested
+several time with neo.rawio (numpy buffer) and neo.io (neo object tree).
+See neo.test.iotest.*
+
+
+Author: Samuel Garcia
```
Comment (samuelgarcia) [inline neo/test/rawiotest/test_mearecrawio.py:21]: Please remove this

File: neo/test/rawiotest/test_mearecrawio.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+from neo.test.rawiotest.common_rawio_test import BaseTestRawIO
+
+
+class TestMEArecRawIO(BaseTestRawIO, unittest.TestCase, ):
+    rawioclass = MEArecRawIO
+    # here obsvisously there is nothing to download:
+    files_to_download = ['mearec_test_10s.h5']
+    # here we will test 2 fake files
+    # not that IO base on dirname you can put the dirname here.
+    entities_to_test = ['mearec_test_10s.h5']
```
Comment (samuelgarcia) [inline neo/test/rawiotest/test_mearecrawio.py:38]: Remove unnecessary comments also here.

**Comment by samuelgarcia** [comment]
Hi Alessio.
A bit of more clean and then we can merge this.
It is OK for me.

**Comment by alejoe91** [comment]
> Hi Alessio.
> A bit of more clean and then we can merge this.
> It is OK for me.

Done :)

**Comment by samuelgarcia** [comment]
This is OK for me.
@JuliaSprenger do you want to have look before I merge this ?

**Comment by samuelgarcia** [comment]
Can I merge this ?

**Pull request merged**